Cython/Compiler/Scanning.py

```diff
@@ -477,13 +481,24 @@ def unread(self, token, value):
         # This method should be added to Plex
         self.queue.insert(0, (token, value))
 
-    def error(self, message, pos=None, fatal=True):
+    def error(self, message, pos=None):
+
+        fatal = not self.fault_tolerant
+
         if pos is None:
             pos = self.position()
         if self.sy == 'INDENT':
-            error(pos, "Possible inconsistent indentation")
-        err = error(pos, message)
-        if fatal: raise err
+            if fatal:
```

Comment:
This should not change. Rather, in the test case set: `DebugFlags.debug_exception_on_error` so that `error(...)` does not raise an exception.

Then, to get the actual errors, I think you can do something as:

```
from Cython.Compiler.Errors import local_errors
with local_errors() as errors:
    parse_from_strings(...)
    # At this point the errors found should be in the `errors` variable, so, we can assert checking those.
```